drafts/AENS.md

```diff
@@ -0,0 +1,414 @@
+
+1. one time fee, which can be claimed by miner
+2. one time fee which gets burned
+3. deposit which can be reclaimed after the name is released
+4. no additional fee other than the standard transaction fee
+
+For the actual amount of fees to be paid we have a couple of options:
+
+***Options:***
+
+1. based on length (e.g. have buckets of different price categories)
+2. every name pays the same amount of fees (or none)
+
+Each entry has a fixed expiration date on claim after which the entry should 
+transition into the `revoked` state. Once it reaches this state, the name 
+will be released, i.e. transition into the `unclaimed` state, after which 
+it could be claimed again.
+
+To prevent the entry from expiring a user can, at any point before reaching
+the expiration date, update the entry, which pushes the expiration date
+into the future by the time delta of the registration time and the update
+time, e.g. if a user posts an update one week after claiming their name,
+the expiration date gets pushed one week further into the future.
+The main motivation of this expiration date is to prevent situations where
+the private keys, which control the name, are lost making the name unusable 
+as well.
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", 
+"SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be 
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+
+### AENS Entry
+
+```
+  name         size (bytes)
+ ------------ ----
+| owner      | 32 |
+ ------------ ----
+| expires_by | 32 |
+ ------------ ----
+| ttl        | 8  |
+ ------------ ----
+| pointers   |    |
+ ------------ ----
+```
+
+***owner***: the account, which controls this entry.
+
+***expires_by***: the blockheight after which the entry goes
+into the `revoked` state.
+
+***ttl***: a suggestion as to how long any clients should 
+cache this information. (***TODO***: should have a reasonable 
+upper limit, e.g. 86400 seconds)
+
+***pointers***: a tuple with all the values this entry
+points towards, e.g. `{kind: "ipfs", data: "QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC"}`
+
+
+### Name
+
+Names MUST be normalized according to IDNA2008 before hashing, as 
+described in [uts-46](https://unicode.org/reports/tr46) and SHOULD 
+be stored in their ASCII representation.
+
+See also [rfc3491](https://tools.ietf.org/html/rfc3491) and 
+[rfc3492](https://tools.ietf.org/html/rfc3492).
+
+Names SHOULD be at most 253 characters long, in order to enable 
+interoperability with DNS.
+
+***Note:*** Fot this first draft names MUST have `.aet` or `.test`
+as a suffix, i.e. `mywallet.aet`.
+
+
+### Namespaces/Labels
+
+A name is split up into labels by the `FULL STOP (U+002E .)` character.
+These labels will also be referred to as namespaces, e.g. `mywallet.aet` 
+can be understood as the `mywallet` namespace in the `aet` namespace.
+
+A label MUST be longer than three characters.
```

Comment:
Longer than or equal to three characters? Or are the top-level namespaces (`aet`) exempt?

Reply:
Longer than

The top-level namespaces will have some stricter rules applied to them once they become available.